packages/react-combobox/src/components/Option/useOption.ts

```diff
@@ -12,17 +29,68 @@ import type { OptionProps, OptionState } from './Option.types';
  * @param ref - reference to root HTMLElement of Option
  */
 export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElement>): OptionState => {
+  const {
+    activeOption,
+    idBase,
+    onOptionClick,
+    registerOption,
+    selectedOptions = [],
+    unRegisterOption,
+  } = useContextSelector(ListboxContext, ctx => ({
+    activeOption: ctx.activeOption,
+    idBase: ctx.idBase,
+    onOptionClick: ctx.onOptionClick,
+    registerOption: ctx.registerOption,
+    selectedOptions: ctx.selectedOptions,
+    unRegisterOption: ctx.unRegisterOption,
+  }));
+  const { id, fluentKey: key, disabled, value } = props;
+  const selected = key ? !!selectedOptions.find(option => option.key === key) : false;
+  const optionValue = getValueString(value, props.children);
+
+  // use the id if provided, otherwise construct id from key & idBase
+  const optionId = id || key ? `${idBase}-${props.fluentKey}` : '';
```

Comment:
we'd also need to use a `useRef` hook to prevent the `id` from updating on an option re-render, which would then create an infinite render loop (b/c the option `id` changing would need the parent to re-render, etc. etc.).

I'd agree we need this if we get rid of `key`, but is there a benefit to calling `useId` within the options now?